### PR TITLE
Added possibility to reorder files in working sets

### PR DESCRIPTION
**Pull request by macdems**
Thank you for the extension. This is what I was missing from the VSC.

I suggest two improvementsThank you for the extension. This is what I was missing from the VSCode.

I suggest two improvements:

* to change the activity bar icon to match the VSCode theme and

* to add possibility to reorder files in each working set manually (not only sort them alphabetically).

This PR adds these improvements. Please consider it for merging with your extension.

### CODE REVIEW

**Comment by bernardop** [comment]
@macdems thank you for taking the time to contribute! This looks very good.

I'd like to keep the icon in the activity bar to be the same as the one in the README of the project. And as far as I understand, user-added svgs are not allowed when you publish the extension. 

Screen Shot 2020-08-04 at 10 55 52 AM

So I created an outlined version of the png icon, do you mind using this instead of the one you added?

briefcase-outline

File: package.json [hunk 1]
```diff
@@ -234,7 +256,9 @@
     "lint": "eslint src --ext ts",
     "watch": "tsc -watch -p ./",
     "pretest": "npm run test-compile && npm run lint",
-    "test": "node ./out/test/runTest.js"
+    "test": "node ./out/test/runTest.js",
+    "package": "vsce package",
+    "publish": "vsce publish"
```
Comment (bernardop) [inline package.json:261]: Could you add a new script below "publish" to run `prettier` on the `src` directory and run it? To keep the formatting consistent.
```
"prettier": "prettier --write src"
```

File: src/types.ts [hunk 2]
```diff
@@ -75,6 +75,15 @@ export class WorkingSet extends vscode.TreeItem {
     })
   }
 
+  moveItem(filePath: string, direction: MoveDirection) {
+    const index = this.items.findIndex(({ resourceUri: { fsPath } }) => fsPath === filePath);
+    if (direction == MoveDirection.UP && index > 0) {
```
Comment (bernardop) [inline src/types.ts:80]: Could you use strict equality here? 
```
    if (direction === MoveDirection.UP && index > 0) {
```

File: src/types.ts [hunk 3]
```diff
@@ -75,6 +75,15 @@ export class WorkingSet extends vscode.TreeItem {
...
+      [this.items[index - 1], this.items[index]] = [this.items[index], this.items[index - 1]]
+    } else if (direction == MoveDirection.DOWN && index != -1 && index < this.items.length - 1) {
```
Comment (bernardop) [inline src/types.ts:82]: Same here
```
    } else if (direction === MoveDirection.DOWN && index !== -1 && index < this.items.length - 1) {
```

File: src/workingSets.ts [hunk 4]
```diff
@@ -338,6 +339,13 @@ export class WorkingSetsProvider
     this.updateWorkspaceState()
   }
 
+  moveFile(workingSetItem: WorkingSetItem, direction: MoveDirection) {
+    if (!workingSetItem) return;
```
Comment (bernardop) [inline src/workingSets.ts:343]: Could you add curly braces around the `return` to keep `eslint` happy 😄? 
```
    if (!workingSetItem) { return; }
```

File: package.json (hunk 1 above)
Comment (macdems) [reply package.json:261]: Sure

File: src/workingSets.ts (hunk 4 above)
Comment (macdems) [reply src/workingSets.ts:343]: Done

**Comment by macdems** [comment]
Thank you for your answer. I have done suggested changes. However, I am not convinced that the restriction of the SVG icons apply here. I understand that the main package icons may not be SVGs (which makes sense as they are displayed BEFORE installing extension, similarly to any image in the `README.md` and `CHANGELOG.md`). You are already using SVG icons anyway.

In any case I have adapted the icon to more closely match your original design, however I kept it consistent with the general VSCode theme (for example, all the lines have the same thickness, which matches well standard activity bar icons). Converted to PNG it looks like this:
briefcase-72

File: src/types.ts [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import * as vscode from "vscode"
 import { basename } from "path"
 import { existsSync } from "fs"
+import { threadId } from "worker_threads"
```
Comment (bernardop) [inline src/types.ts:4]: Did this get added by accident?

**Comment by bernardop** [comment]
That new icon looks great! Thanks @macdems! And you may be right, I may be misinterpreting that svg rule. I guess we'll find out when I publish the update 😅.

File: src/types.ts (hunk 5 above)
Comment (macdems) [reply src/types.ts:4]: I did not type it. Some tool must have added it. Removing...

**Comment by bernardop** [comment]
Merged, I'll publish a new version to the marketplace later today. Thanks @macdems!

**Comment by macdems** [comment]
:+1:

**Pull request merged**